skiko/src/awtMain/kotlin/org/jetbrains/skiko/swing/SkiaSwingLayer.kt

```diff
@@ -115,6 +115,8 @@ open class SkiaSwingLayer(
                 redrawerManager.findNextWorkingRenderApi()
                 repaint()
             }
+        } catch (e: Exception) {
+            Logger.warn(e) { "Exception during SkiaSwingLayer rendering" }
```

Comment:
@m-sasha I agree, but I would say that this part of the framework is crucial, so not handling exceptions here would cause completely broken view (like in attached issue)

Reply:
But why would there be exceptions here? Is it not better to discover and fix them instead?
In Compose we catch exceptions at the window level and typically show an error dialog followed by shutting down the application.

Reply:
They can appear only because of the broken internal contracts (out mistakes) and for sure they should be fixed. Like here https://github.com/JetBrains/compose-multiplatform-core/pull/878

Reply:
Then we should introduce this policy for swing bridge then too. Do you think it would be better?

Reply:
I would recommend that, yes. I wonder what @igordmn thinks.

Reply:
We had similar code in the past (try-catch-log for rendering), and it was bad UX for end users (the window just stops to draw). We moved exception handling responsibility to `SkikoView`, as exceptions happen on its level, and it knows what to do with it.

In case of Swing components (which ComposePanel is), what is the policy for exceptions during rendering?

Regarding what policy we choose, we should implement it on the `ComposeBridge` level. Actually it is already implemented (see `ComposePanel.exceptionHandler`), we just need to define the default behavior.

Reply:
> In case of Swing components (which ComposePanel is), what is the policy for exceptions during rendering?

Looks like it just calls the thread's uncaught exception handler (from `EventDispatchThread.pumpOneEventForFilters`).

Reply:
If that is the case, it seems everything already works this way?

Reply:
Only by default (which I also think is a bad idea). In Swing I can set the thread's uncaught exception handler and change the behavior.